src/main/java/org/sefglobal/scholarx/service/CommentService.java

```diff
@@ -0,0 +1,102 @@
+    private final ProfileRepository profileRepository;
+    private final MenteeRepository menteeRepository;
+
+    public CommentService(ProfileRepository profileRepository,MenteeRepository menteeRepository,
+                                CommentRepository commentRepository){
+        this.commentRepository = commentRepository;
+        this.profileRepository = profileRepository;
+        this.menteeRepository = menteeRepository;
+    }
+
+    public List<Comment> getAllMenteeComments(long menteeId, long profileId)
+            throws ResourceNotFoundException, BadRequestException, UnauthorizedException {
+        Optional<Mentee> optionalMentee = menteeRepository.findById(menteeId);
+
+        if (!optionalMentee.isPresent()) {
+            String msg = "Error, Mentee by id: " + menteeId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        } else if (optionalMentee.get().getProfile().getId() == profileId) {
+            String msg = "Error, User by id: " + menteeId + " is not allowed access.";
+            log.error(msg);
+            throw new UnauthorizedException(msg);
+        }
+
+        return commentRepository.findAllByMenteeId(menteeId);
+    }
+
+    public Comment addMenteeComment(long menteeId, long profileId, String menteeComment)
+            throws ResourceNotFoundException, UnauthorizedException {
+        Optional<Profile> optionalProfile = profileRepository.findById(profileId);
+        Optional<Mentee> optionalMentee = menteeRepository.findById(menteeId);
+
+        if (!optionalMentee.isPresent()) {
+            String msg = "Error, Mentee by id: " + menteeId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+
+        if (!optionalProfile.isPresent()) {
+            String msg = "Error, User by id: " + profileId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        } else if (optionalMentee.get().getProfile().getId() == profileId) {
+            String msg = "Error, User by id: " + menteeId + " is not allowed access.";
+            log.error(msg);
+            throw new UnauthorizedException(msg);
+        }
+
+        Comment comment = new Comment();
+        comment.setProfile(optionalProfile.get());
+        comment.setComment(menteeComment);
+        comment.setMentee(optionalMentee.get());
+        return commentRepository.save(comment);
+    }
+
+    public Comment updateComment(long id, long profileId, String menteeComment)
+            throws ResourceNotFoundException {
+        Optional<Comment> optionalComment = commentRepository.findById(id);
+        if (!optionalComment.isPresent()) {
+            String msg = "Error, Comment with id: " + id + " cannot be deleted. " +
+                    "Comment doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
```

Comment:
Check whether the profile id on comment is equal to the profile id taken from authentication

Reply:
ok cool